internal/bus/bus_redis.go

```diff
@@ -73,33 +73,33 @@ func NewRedisMessageBus(rc redis.UniversalClient) MessageBus {
 	return r
 }
 
-func (r *redisMessageBus) Publish(_ context.Context, channel string, msg proto.Message) error {
+func (r *redisMessageBus) Publish(_ context.Context, channel Channel, msg proto.Message) error {
 	b, err := serialize(msg)
 	if err != nil {
 		return err
 	}
 
 	r.mu.Lock()
-	ops, ok := r.publishOps[channel]
+	ops, ok := r.publishOps[channel.Primary]
 	if !ok {
 		ops = &redisWriteOpQueue{}
-		r.publishOps[channel] = ops
+		r.publishOps[channel.Primary] = ops
 	}
-	ops.push(&redisPublishOp{r, channel, b})
+	ops.push(&redisPublishOp{r, channel.Primary, b})
 	r.mu.Unlock()
 
 	if !ok {
-		r.enqueueWriteOp(&redisExecPublishOp{r, channel, ops})
+		r.enqueueWriteOp(&redisExecPublishOp{r, channel.Primary, ops})
 	}
 	return nil
 }
 
-func (r *redisMessageBus) Subscribe(ctx context.Context, channel string, size int) (Reader, error) {
-	return r.subscribe(ctx, channel, size, r.subs, false)
+func (r *redisMessageBus) Subscribe(ctx context.Context, channel Channel, size int) (Reader, error) {
+	return r.subscribe(ctx, channel.Primary, size, r.subs, false)
 }
 
-func (r *redisMessageBus) SubscribeQueue(ctx context.Context, channel string, size int) (Reader, error) {
-	return r.subscribe(ctx, channel, size, r.queues, true)
+func (r *redisMessageBus) SubscribeQueue(ctx context.Context, channel Channel, size int) (Reader, error) {
+	return r.subscribe(ctx, channel.Primary, size, r.queues, true)
```

Comment:
Is there a need to do the migration from `legacy` -> `primary` for redis?

Reply:
perhaps not.. we could make subject construction specific to each bus

Reply:
we should do it anyway so the implementations are aligned to keep it simple.